client/packages/invoices/src/Returns/modals/OutboundReturn/ItemSelector.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { FC } from 'react';
+import {
+  ModalRow,
+  ModalLabel,
+  Grid,
+  useTranslation,
+  BasicTextInput,
+} from '@openmsupply-client/common';
+import {
+  ItemStockOnHandFragment,
+  StockItemSearchInput,
+} from '@openmsupply-client/system';
+import { useReturns } from '../../api';
+
+interface ItemSelectorProps {
+  item: ItemStockOnHandFragment | null;
+  disabled: boolean;
+  onChangeItem: (item: ItemStockOnHandFragment) => void;
+}
+
+export const ItemSelector: FC<ItemSelectorProps> = ({
+  item,
+  disabled,
+  onChangeItem,
+}) => {
+  const t = useTranslation();
+
+  const { data } = useReturns.document.outboundReturn();
+
+  const existingItemIds = data?.lines.nodes.map(line => line.itemId);
+
+  return (
+    <>
+      <ModalRow>
+        <ModalLabel
+          label={t('label.item', { count: 1 })}
+          justifyContent="flex-end"
+        />
+        <Grid item flex={1}>
+          <StockItemSearchInput
+            autoFocus={!item}
+            openOnFocus={!item}
+            disabled={disabled}
+            currentItemId={item?.id}
+            onChange={newItem => newItem && onChangeItem(newItem)}
+            extraFilter={
+              disabled
+                ? undefined
+                : item =>
+                    item.availableStockOnHand !== 0 &&
+                    !existingItemIds?.some(id => id === item.id)
+            }
+          />
+        </Grid>
+      </ModalRow>
+      {item && (
+        <ModalRow margin={3}>
+          <ModalLabel label={t('label.unit')} justifyContent="flex-end" />
+          <BasicTextInput
+            disabled
+            sx={{ width: 150, marginBottom: '10px' }}
+            value={item.unitName ?? ''}
```

Comment:
You shouldn't need the fallback `?? ''` since the item must be defined here (I tried removing it and it seemed fine)